Formula/vile.rb

```diff
@@ -0,0 +1,27 @@
+class Vile < Formula
+  desc "Vi Like Emacs Editor"
+  homepage "https://invisible-island.net/vile/"
+  url "https://invisible-island.net/archives/vile/current/vile-9.8v.tgz"
+  sha256 "240edec7bbf3d9df48b3042754bf9854d9a233d371d50bba236ec0edd708eed5"
+  license "GPL-2.0-or-later"
+
+  depends_on "groff" => :build
+  uses_from_macos "flex" => :build
+
+  uses_from_macos "libiconv"
+  uses_from_macos "ncurses"
+
+  def install
+    system "./configure", "--prefix=#{prefix}",
+                          "--disable-imake",
+                          "--enable-colored-menus",
+                          "--with-ncurses",
+                          "--with-screen=ncurses",
+                          "--without-x"
+    system "make", "install"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I'll try to make it process a file and check the output. I'll update it today.

Reply:
Fixed in https://github.com/Homebrew/homebrew-core/pull/106086/commits/89a13002c1bd3cde72283a672f667c23fa608b37#diff-3fb6d8265767b629e5c8862db30f7a88d664cd63a86ec9ebfae31ff209063b83R31 - now it uses expect to start the editor, create a new file, quit cleanly, and ensure the file was created.